src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -46,9 +46,13 @@ export async function searchLogGroup(registry: LogStreamRegistry): Promise<void>
         }
 
         await registry.registerLog(uri, initialStreamData)
-        const doc = await vscode.workspace.openTextDocument(uri) // calls back into the provider
-        vscode.languages.setTextDocumentLanguage(doc, 'log')
-        await vscode.window.showTextDocument(doc, { preview: false })
+
+        if (registry.hasLog(uri)) {
+            // If the log recieved data i.e. wasn't cancelled.
```

Comment:
Notice how much more complicated our code got by adding very basic cancellation logic. And we haven't even gotten to testing it yet.

These comments are sign of a bigger problem. Our command can now silently fail without the user having any clue what just happened because we've added a lot of assumptions. Why even let this code execute if `registerLog` was cancelled?

Reply:
I see what you mean. I am assuming that the log will be `deRegistered` only if the user cancels. The only way around this I could think of was the give the response a `cancelled` attribute that we can directly check before running this code. That way we make no assumptions. However, my implementation feels a little clunky, so not sure if this is best approach or if I just implemented it wrong.